File: app/src/main/java/com/example/acplite/adapters/RecyclerEventsAdapter.java
package com.example.acplite.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.acplite.R;
import com.example.acplite.entidades.Evento;
import com.example.acplite.entidades.Productos;
import com.example.acplite.helpers.ItemTapListener;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

public class RecyclerEventsAdapter extends RecyclerView.Adapter<RecyclerEventsAdapter.ViewHolder> {

    private ArrayList<Evento> eventList;
    private final ItemTapListener mTapListener;

    public RecyclerEventsAdapter(ArrayList<Evento> eventList, ItemTapListener mTapListener) {
        this.eventList = eventList;
        this.mTapListener = mTapListener;
    }

    @NonNull
    @NotNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {
        RecyclerEventsAdapter.ViewHolder viewHolder = new RecyclerEventsAdapter.ViewHolder(
                LayoutInflater.from(parent.getContext()).inflate(R.layout.item_events, parent, false),
                mTapListener);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull @NotNull ViewHolder holder, int position) {
        holder.txtEventName.setText(eventList.get(position).getEventName());
        holder.txtEventDate.setText("Fecha: " + eventList.get(position).getEventDate());
    }

    @Override
    public int getItemCount() {
        return eventList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        TextView txtEventName, txtEventDate;

        public ViewHolder(@NonNull @NotNull View itemView, ItemTapListener itemTapListener) {
            super(itemView);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    mTapListener.onItemTap(v, getAdapterPosition());
                }
            });

            txtEventName = itemView.findViewById(R.id.txtEventName);
            txtEventDate = itemView.findViewById(R.id.txtEventDate);

        }
    }
}

File: app/src/main/java/com/example/acplite/adapters/RecyclerEventsAdapter.kt
package com.example.acplite.adapters

import com.example.acplite.entidades.Evento
import com.example.acplite.helpers.ItemTapListener
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import com.example.acplite.R
import android.widget.TextView
import java.util.ArrayList

class RecyclerEventsAdapter(
    private val eventList: ArrayList<Evento>,
    private val mTapListener: ItemTapListener
) : RecyclerView.Adapter<RecyclerEventsAdapter.ViewHolder>() {
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        return ViewHolder(
            LayoutInflater.from(parent.context).inflate(R.layout.item_events, parent, false),
            mTapListener
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.txtEventName.text = eventList[position].eventName
        holder.txtEventDate.text = "Fecha: " + eventList[position].eventDate
    }

    override fun getItemCount(): Int {
        return eventList.size
    }

    inner class ViewHolder(itemView: View, itemTapListener: ItemTapListener?) :
        RecyclerView.ViewHolder(itemView) {
        var txtEventName: TextView
        var txtEventDate: TextView

        init {
            itemView.setOnClickListener { v -> mTapListener.onItemTap(v, adapterPosition) }
            txtEventName = itemView.findViewById(R.id.txtEventName)
            txtEventDate = itemView.findViewById(R.id.txtEventDate)
        }
    }
}